multiplexer/abci/multiplexer.go

```diff
@@ -624,43 +535,93 @@ func (m *Multiplexer) startCmtNode() error {
 		return err
 	}
 
-	if err := tmNode.Start(); err != nil {
+	if err := cmNode.Start(); err != nil {
 		return err
 	}
 
-	m.registerCleanupFn(func() error {
-		if tmNode != nil && tmNode.IsRunning() {
-			return tmNode.Stop()
-		}
-		return nil
-	})
+	m.cmNode = cmNode
+	return nil
+}
+
+// Stop stops the multiplexer and all its components.
+func (m *Multiplexer) Stop() error {
+	m.logger.Info("stopping multiplexer")
+	if err := m.stopCometNode(); err != nil {
```

Comment:
for example, if you click ctrl-c, the sig notifier will call Stop(), then the stop errors at this level. The signals notifier keeps listening for subsequent ctrl-c? 

also, if a user click ctrl-c, I assume he wants to exit, does it make sense to return an error and not stop the rest? or try to stop as many services as possible while logging the errors?